File: phpunit/tests/SessionTest.php
<?php

use PHPUnit\Framework\TestCase;

final class SessionTest extends TestCase
{
    public function testRefreshesATokenThatIsAboutToExpire(): void
    {
        if (is_first_attempt()) {
            throw new RuntimeException('Timeout: refresh did not complete within 5000ms');
        }
        $this->assertTrue(true);
    }

    public function testRejectsATokenSignedWithTheWrongKey(): void
    {
        $this->expectException(InvalidArgumentException::class);
        throw new InvalidArgumentException('invalid signature');
    }
}

File: gradle/src/test/java/lab/SessionTest.java
package lab;

import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class SessionTest {

    @Test
    void refreshesATokenThatIsAboutToExpire() {
        if (Attempt.isFirst()) {
            throw new IllegalStateException("Timeout: refresh did not complete within 5000ms");
        }
        assertTrue(true);
    }

    @Test
    void rejectsATokenSignedWithTheWrongKey() {
        assertThrows(IllegalArgumentException.class, () -> {
            throw new IllegalArgumentException("invalid signature");
        });
    }
}
